File: reader/server/core/spring/src/test/kotlin/com/richodemus/reader/reader/test/pages/model/FeedId.java
package com.richodemus.reader.reader.test.pages.model;

public class FeedId {
    private final String value;

    public FeedId(final String value) {
        this.value = value;
    }

    public String toJson() {
        return "\"" + value + "\"";
    }

    @Override
    public String toString() {
        return value;
    }
}

File: dto/src/main/kotlin/com/richodemus/reader/dto/FeedId.kt
package com.richodemus.reader.dto

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonValue

data class FeedId(@get:JsonIgnore val value: String) {
    init {
        require(value.isNotBlank()) { "FeedId can't be empty" }
        Pair(1, 2) // This is just here so stdlib is used for something...
    }

    @JsonValue fun getId() = value

    override fun toString() = value
}
